File: bindinterface-compiler/src/main/java/net/chmielowski/bindinterface/FileFactory.java
package net.chmielowski.bindinterface;

import java.io.File;
import java.util.Objects;

import javax.annotation.processing.ProcessingEnvironment;

class FileFactory {

    private static ProcessingEnvironment environment;

    static void initialize(final ProcessingEnvironment env) {
        environment = env;
    }

    static File newKotlinFile() {
        final String directory = Objects.requireNonNull(
                environment.getOptions().get("kapt.kotlin.generated"),
                "Can't find the target directory for generated Kotlin files."
        );
        return new File(directory);
    }
}

File: bindinterface-compiler/src/main/java/net/chmielowski/bindinterface/FileFactory.kt
package net.chmielowski.bindinterface

import java.io.File
import java.lang.Exception
import java.util.Objects

import javax.annotation.processing.ProcessingEnvironment

internal object FileFactory {

    private var environment: ProcessingEnvironment? = null

    @JvmStatic
    fun initialize(env: ProcessingEnvironment) {
        environment = env
    }

    @JvmStatic
    fun newKotlinFile(): File {
        val env: ProcessingEnvironment = environment ?: throw Exception("ProcessingEnvironment not initialized.")
        val directory: String = env.options["kapt.kotlin.generated"]
            ?: throw Exception("Can't find the target directory for generated Kotlin files.")
        return File(directory)
    }
}
